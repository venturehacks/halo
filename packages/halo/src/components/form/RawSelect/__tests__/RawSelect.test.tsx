import { render } from 'enzyme';
import React from 'react';

import { RawSelect, RawSelectOption } from '../index';

const options: RawSelectOption[] = [
  { label: 'Chocolate', value: 'chocolate' },
  { label: 'Vanilla', value: 'vanilla' },
  { label: 'Strawberry', value: 'strawberry' },
];

describe('RawSelect', () => {
  test('smoke', () => {
    const component = render(<RawSelect options={options} />);

    expect(component.text()).toContain('ChocolateVanillaStrawberry');
  });

  describe('snapshot', () => {
    describe('size', () => {
      test('sm', () => {
        const component = render(<RawSelect options={options} size="sm" />);
        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = render(<RawSelect options={options} />);
        expect(component).toMatchSnapshot();
      });

      test('lg', () => {
        const component = render(<RawSelect options={options} size="lg" />);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
