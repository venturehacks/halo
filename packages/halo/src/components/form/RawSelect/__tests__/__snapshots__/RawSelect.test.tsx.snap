// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RawSelect snapshot size lg 1`] = `
<select
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a p-3 h-12 bg-white rounded focus:border-2 border-slate-500 border disabled:text-dark-a disabled:bg-slate-200 disabled:border-light-aa w-auto"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
</select>
`;

exports[`RawSelect snapshot size md 1`] = `
<select
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a px-3 py-2 h-10 bg-white rounded focus:border-2 border-slate-500 border disabled:text-dark-a disabled:bg-slate-200 disabled:border-light-aa w-auto"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
</select>
`;

exports[`RawSelect snapshot size sm 1`] = `
<select
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a px-3 py-1 h-8 bg-white rounded focus:border-2 border-slate-500 border disabled:text-dark-a disabled:bg-slate-200 disabled:border-light-aa w-auto"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
</select>
`;
